x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/authz/ReadActionsTests.java

```diff
@@ -223,6 +224,38 @@ public void testExplicitAndWildcardInclusionAndExplicitExclusions() {
         assertReturnedIndices(trySearch("test10", "test111", "test112", "-test111", "-test112", "index*"), "test10");
     }
 
+    public void testEmptyExpressionIsInvalid() {
+        createIndicesWithRandomAliases("test1");
+
+        final List<String[]> expressionsList = List.of(
+            new String[] { "" },
+            new String[] { "*", "" },
+            new String[] { "test1", "" },
+            new String[] { "*", "", "test1" }
+        );
+        for (var expressions : expressionsList) {
+            final var e = expectThrows(ElasticsearchSecurityException.class, () -> trySearch(expressions).get());
+            assertThat(e.getCause(), instanceOf(IllegalArgumentException.class));
```

Comment:
`IllegalArgumentException` is always [wrapped](https://github.com/elastic/elasticsearch/blob/0cd9c72199e3b665be6e816993647439e962ee21/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationService.java#L597-L630) with `ElasticsearchSecurityException`. I think we could argue whether such wrapping is necessary. But I don't want to go down that path for now. Alternatively, if we want to avoid the wrapping, we could throw either `InvalidIndexNameException` or `IndexNotFoundException` instead. Please let me know if you prefer either of these.

Reply:
On second thought, I decided to use `InvalidIndexNameException` which seems more specific for the particular issue. eb9498bbf216770149ff2817e32e0bd435e21f7e